app.py

```diff
@@ -472,6 +509,49 @@ def invoice():
     #TODO: other statuses..
     return render_template("invoice.html", invoice=invoice, order=order, error=error, qrcode_svg=qrcode_svg, url=url)
 
+@app.route('/bronze_oauth')
+def bronze_oauth():
+    if not bronze.authorized:
+        return redirect(url_for('bronze.login'))
+    res = ''
+    validate = bronze_blueprint.session.get('Validate')
+    if not validate.ok:
+        return 'oauth access token not able to validate'
+    userinfo = bronze_blueprint.session.get('UserInfo')
+    if userinfo.ok:
+        userinfo_content = userinfo.json()
+        res += '<p>Your userinfo is {}</p>'.format(userinfo.content)
+    else:
+        res += '<h1>Failed to get userinfo</h1>'
+    #balance = bronze_blueprint.session.get('AccountBalance')
+    #if balance.ok:
+    #    balance_content = balance.json()
+    #    res += '<p>Your balance is {}</p>'.format(balance.content)
+    #else:
+    #    res += '<h1>Failed to get balance</h1>'
+    kyc = bronze_blueprint.session.get('AccountKyc')
+    if kyc.ok:
+        kyc_info = kyc.json()
+        level = kyc_info['level']
+        if not level < '2':
+            res+= 'KYC level(acceptable): {}'.format(level)
+        else:
+
+            res+= 'KYC level(unacceptable): {}'.format(level)
+    else:
+        res += '<h1>Failed to get kyc</h1>'
+    #print(res)
+    url = app.config["BRONZE_ADDRESS"] + '/Manage/Kyc'
+    utilities = Utility.all_alphabetical(db.session)
+    return render_template('utilities.html', utilities=utilities, bronze_auth=bronze.authorized, kyc_info=kyc_info, userinfo=userinfo_content, url=url)
+    #return render_template('utilities.html', utilities=utilities, bronze_auth=bronze.authorized, balance_info=balance_content, kyc_info=kyc_info, userinfo=userinfo_content, url=url)
+    
+@app.route("/bronze_logout")
+def logout():
+    del bronze_blueprint.token
```

Comment:
can bronze_blueprint.token  not exist?

Reply:
change to place it inside an if condition.